client/src/commands/instantiateCommand.ts

```diff
@@ -157,6 +164,9 @@ export async function instantiateSmartContract(treeItem?: BlockchainTreeItem): P
             if (packageEntry) {
                 smartContractName = packageEntry.name;
                 smartContractVersion = packageEntry.version;
+            } else if (chaincodeContainerName) {
```

Comment:
I think we should move this whole if else block higher up to where you first choose the smart contract. You wouldn't need so many ifs then